### PR TITLE
Add AddDoctorCommand

### PR DESCRIPTION
**Pull request by vigonometry**
Added AddDoctorCommand class that supports the creation of doctor entities in AddressBook.

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -79,6 +81,36 @@ public static Email parseEmail(String email) throws ParseException {
...
+    /**
+     * Parses a {@code String yoe} into an {@code Yoe}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code yoe} is invalid.
+     */
+    public static Yoe parseYoe(String yoe) throws ParseException {
+        requireNonNull(yoe);
+        String trimmedYoe = yoe.trim();
+        if (!Specialty.isValidSpecialty(trimmedYoe)) {
```
Comment (praveenkrishna0512) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:108]: This should be `(!Yoe.isValidYoe(trimmedYoe)`

File: src/test/java/seedu/address/logic/commands/AddCommandTest.java [hunk 2]
```diff
@@ -113,6 +114,16 @@ public void addPerson(Person person) {
             throw new AssertionError("This method should not be called.");
         }
 
+        @Override
+        public void addDoctor(Doctor doctor) {
```
Comment (praveenkrishna0512) [inline src/test/java/seedu/address/logic/commands/AddCommandTest.java:118]: May I clarify if the assertion error is a placeholder for now?

**Review by praveenkrishna0512 — CHANGES_REQUESTED** [review]
Looks good overall! Just require one bug fix :)

File: src/test/java/seedu/address/logic/commands/AddCommandTest.java (hunk 2 above)
Comment (vigonometry) [reply src/test/java/seedu/address/logic/commands/AddCommandTest.java:118]: Yes, it is a placeholder for now. Test cases will be written for AddDoctor command once the initial set of functions is complete.

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 1 above)
Comment (vigonometry) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java]: Resolved

File: src/main/java/seedu/address/logic/commands/AddDoctorCommand.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            + PREFIX_EMAIL + "EMAIL "
+            + PREFIX_SPECIALITY + "SPECIALITY "
+            + PREFIX_YOE + "YEARS OF EXPERIENCE "
+            + "[" + PREFIX_TAG + "TAG]...\n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_NAME + "John Doe "
+            + PREFIX_PHONE + "98765432 "
+            + PREFIX_EMAIL + "johnd@example.com "
+            + PREFIX_SPECIALITY + "Cardiology "
+            + PREFIX_TAG + "Head of Department ";
```
[the remark this answers is not in the document]
Comment (vigonometry) [reply src/main/java/seedu/address/logic/commands/AddDoctorCommand.java:36]: Added

**Review by praveenkrishna0512 — APPROVED** [review]
LGTM!

**Pull request merged**